tests/main/cgroup-devices-v2/task.yaml

```diff
@@ -0,0 +1,112 @@
+    # ready. As a workaround, touch a "remove-me" file that is removed by the
+    # service on startup, restart the service and the wait for the file to
+    # disappear.
+    mkdir -p /var/snap/test-snapd-service/common
+    touch /var/snap/test-snapd-service/common/remove-me
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-service
+    # Wait for the service to really be alive and running. Otherwise the "main pid"
+    # will be still tracking snap-run-confine-exec chain and be unreliable.
+    retry -n 5 test ! -e /var/snap/test-snapd-service/common/remove-me
+
+    echo "Extract the PID of the main process tracked by systemd"
+    # It would be nicer to use "systemctl show --property=... --value" but it doesn't work on older systemd.
+    pid=$(systemctl show snap.test-snapd-service.test-snapd-service.service --property=ExecMainPID | cut -d = -f 2)
+
+    echo "Extract the cgroup of the main process"
+    cgroup=$(grep 0:: < "/proc/$pid/cgroup" | cut -d : -f 3)
+
+    # The service always runs in a device cgroup created by systemd (or in a
+    # scope created snap run for user processes)
+    test "$cgroup" = /system.slice/snap.test-snapd-service.test-snapd-service.service
+
+    # Initially since there are no devices assigned to snap, there is no map with device access
+    test ! -e /sys/fs/bpf/snap/snap_test-snapd-service_test-snapd-service
+    # Do another dump
+    dump_cgroup_device_progs > cgroup_device.progs-2
+    echo "Verify no new device cgroup programs are present"
+    test "$(wc -l < cgroup_device.progs-1)" = "$(wc -l < cgroup_device.progs-2)"
+
+    echo "Connect the joystick interface"
+    snap connect test-snapd-service:joystick
+
+    # a device cgroup is not added until the service gets restarted
+    test ! -e /sys/fs/bpf/snap/snap_test-snapd-service_test-snapd-service
+
+    echo "Run /bin/true via snap-confine, so that we create the device cgroup made by snapd"
+    snap run test-snapd-service.sh -c /bin/true
+    # with new devices assigned to the service, we see a map listing the devices
+    # that access is allowed to
+    echo "Verify that device access map for the application is present and is pinned to a file"
+    test -e /sys/fs/bpf/snap/snap_test-snapd-service_sh
+    # but there are no new programs, as they go away when cgroup is removed
+    # (also give systemd some time to clean up the cgroup)
+    sleep 1
+    dump_cgroup_device_progs > cgroup_device.progs-3
+    test "$(wc -l < cgroup_device.progs-1)" = "$(wc -l < cgroup_device.progs-3)"
+
+    # dump the map
+    bpftool map dump pinned /sys/fs/bpf/snap/snap_test-snapd-service_sh > device-access.map-app
+
+    # NOTE: the actual permissions may drift over time. We just care about the fact
+    # that there *are* some constraints here now and there were none before.
+    echo "Verify the constraints in device cgroup imposed by snapd on the snap application"
+    MATCH 'key: 63 01 00 00 00 03 00 00 +00 +value: 01' < device-access.map-app
```

Comment:
Would be nice to explain a bit why this particular values are matched (if it's not too hard to explain/hint at)